resolves #26 fold intrinsic attributes reference into document attributes reference (#27)
asciidoctor/asciidoc-docs: modules/attributes/pages/document-attributes-reference.adoc

> The effective value will not be visible to an attribute reference.

Reviewer: If it works exactly the same I think we should use the same wording, so either:

"The effective value can't be resolved using an attribute reference." or "An implied value will not be visible to an attribute reference".

Otherwise, it implies that there's a slight difference.

(sorry not really related to this change but I just noticed it)

Author: There is a slight difference. In the case of an effective value, the attribute reference will see the blank value. In the case of an implied value, the attribute reference won't resolve at all. But I can still find a way to explain this better.

> The effective value will not be visible to an attribute reference.

Author: ```suggestion
An attribute reference will resolve to an empty value rather than the effective value.
```

If an attribute is not marked as _Header Only_, it can be set anywhere in the document, assuming the attribute is not locked by the API or CLI.
However, changing an attribute only affects behavior for content that follows the assignment (in document order).

== Intrinsic attributes

Intrinsic attributes are set automatically by the processor.
These attributes provide information about the document being processed (e.g., `docfile`), the security mode under which the processor is running (e.g., `safe-mode-name`), and information about the user's environment (e.g., `user-home`).

Many of these attributes are read only, which means they can't be modified (with some exceptions).
Attributes which are not are marked as modifiable.
Attributes marked as both modifiable and API/CLI Only can only be set from the API or CLI.

> All other attributes marked as modifiable must be set by the end of the header (i.e., Header Only).

Reviewer: Maybe we should tell "in the document header" instead of "by the end of the header". The term "Document Header" seems well defined so it might be easier to understand if we use "in the document header".

Author: I've been introducing the phrase "by the end of the header" since that more accurately describes what's really going on. We can't just say "Document Header" because that implies you must set it in the Document Header itself, which is not accurate.

|localdate
|_date (ISO)_ +
_ex._ 2019-02-17
|{y}
|{n}
|Date when the document was converted.^[<<note-sourcedateepoch,2>>]^

|localdatetime
|_datetime (ISO)_ +
_ex._ 2019-02-17 19:31:05 UTC
|{y}
|{n}
|Date and time when the document was converted.^[<<note-sourcedateepoch,2>>]^

|localtime
|_time (ISO)_ +
_ex._ 19:31:05 UTC
|{y}
|{n}
|Time when the document was converted.^[<<note-sourcedateepoch,2>>]^

|localyear
|_integer_ +
_ex._ {localyear}
|{y}
|{n}
|Year when the document was converted.^[<<note-sourcedateepoch,2>>]^

|outdir
|_directory path_ +
_ex._ /home/user/docs/dist
|{n}
|n/a
|Full path of the output directory.
(Cannot be referenced in the content.
Only available to the API once the document is converted).

|outfile
|_file path_ +
_ex._ /home/user/docs/dist/userguide.html
|{n}
|n/a
|Full path of the output file.
(Cannot be referenced in the content.
Only available to the API once the document is converted).

|outfilesuffix
|_file extension_ +
_ex._ .html
|{y}
|{n}
|File extension of the output file (starting with a period) as determined by the backend (`.html` for `html`, `.xml` for `docbook`, etc.).

|safe-mode-level
|`0` +

> `10` +

Author: ```suggestion
`1` +
```

> `20` +

Author: ```suggestion
`10` +
```

> `30`

Reviewer: It should be `0`, `1`, `10` and `20` no?

Author: Good catch!

> `30`

Author: ```suggestion
`20`
```
